testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestCreateDropSchema.java

```diff
@@ -39,18 +42,159 @@
+
+        hdfsClient.delete(externalFile);
+    }
+
+    // Tests create/drop schema transactions with default schema location
+    @Test
+    public void testDropSchemaFilesTransactions()
+    {
+        String schemaName = schemaName("schema_directory_transactions");
+        String schemaDir = format("%s/%s.db/", warehouseDirectory, schemaName);
+
+        createSchema(schemaName);
+        assertFileExists(schemaDir, true, "schema directory exists after create schema");
+
+        onTrino().executeQuery("START TRANSACTION");
+        assertQuerySucceeds("DROP SCHEMA " + schemaName);
+        assertQuerySucceeds("ROLLBACK");
+        assertFileExists(schemaDir, true, "schema directory exists after rollback");
+
+        // Sanity check: schema is still working
+        onTrino().executeQuery(format("CREATE TABLE %s.test_table (i integer)", schemaName));
+        onTrino().executeQuery(format("DROP TABLE %s.test_table", schemaName));
+
+        onTrino().executeQuery("START TRANSACTION");
+        assertQuerySucceeds("DROP SCHEMA " + schemaName);
+        assertQuerySucceeds("COMMIT");
+        assertFileExists(schemaDir, false, "schema directory exists after drop schema");
+    }
+
+    @Test
+    public void testDropSchemaFilesTransactionsWithExternalFiles()
+    {
+        String schemaName = schemaName("schema_transactions_with_external_files");
+        String schemaDir = warehouseDirectory + "/schema-transactions-with-external-files/";
+
+        // Create file in schema directory before creating schema
+        String externalFile = schemaDir + "external-file";
+        hdfsClient.createDirectory(schemaDir);
+        hdfsClient.saveFile(externalFile, "");
+
+        createSchema(schemaName, schemaDir);
+
+        onTrino().executeQuery("START TRANSACTION");
+        assertQuerySucceeds("DROP SCHEMA " + schemaName);
+        assertQuerySucceeds("ROLLBACK");
+        assertFileExists(externalFile, true, "external file exists after rolling back drop schema");
+
+        // Sanity check: schema is still working
+        onTrino().executeQuery(format("CREATE TABLE %s.test_table (i integer)", schemaName));
+        onTrino().executeQuery(format("DROP TABLE %s.test_table", schemaName));
+
+        onTrino().executeQuery("START TRANSACTION");
+        assertQuerySucceeds("DROP SCHEMA " + schemaName);
+        assertQuerySucceeds("COMMIT");
+        assertFileExists(externalFile, true, "schema directory exists after committing drop schema");
+    }
+
+    private void assertFileExists(String path, boolean exists, String description)
+    {
+        assertThat(hdfsClient.exist(path)).as("%s (%s)", description, path).isEqualTo(exists);
+    }
+
+    private static void assertQuerySucceeds(String query)
+    {
+        try {
+            onTrino().executeQuery(query);
+        }
+        catch (QueryExecutionException e) {
+            fail(format("Expected query to succeed: %s", query), e.getCause());
```

Comment:
inline. i.e. replace with plain `onTrino().executeQuery(...)`.

Reply:
I feel like there should be some way to distinguish between a failure in a query assertion vs. a setup query from the logs, and inlining `executeQuery` makes the failures look the same.

Reply:
That's a valid point. But that's also a general statement, not specific to this particular test nor test scenario, and we decided to use `onTrino().executeQuery` indiscriminately for both setup and test assertions.